internal/ingress/metric/controller/metrics.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/ingress-nginx/internal/ingress"
+	"k8s.io/ingress-nginx/internal/ingress/metric"
+)
+
+const (
+	sslLabelExpire = "ssl_expire_time_seconds"
+)
+
+var (
+	operation    = []string{"namespace", "class"}
+	sslLabelHost = []string{"namespace", "class", "host"}
+)
+
+// Metric defines base metrics about the ingress controller
+type Metric struct {
+	prometheus.Collector
+
+	configSuccess         prometheus.Gauge
+	configSuccessTime     prometheus.Gauge
+	reloadOperation       *prometheus.CounterVec
+	reloadOperationErrors *prometheus.CounterVec
+	sslExpireTime         *prometheus.GaugeVec
+
+	namespace string
+	class     string
+}
+
+// NewMetric creates a new Metric.
+func NewMetric(namespace, class string) *Metric {
+	constLabels := prometheus.Labels{
+		"namespace": namespace,
+		"class":     class,
+	}
+
+	cm := &Metric{
+		namespace: namespace,
+		class:     class,
+
+		configSuccess: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace:   metric.Namespace,
+				Name:        "config_last_reload_successfull",
+				Help:        "Whether the last configuration reload attemp was successful",
+				ConstLabels: constLabels,
+			}),
+		configSuccessTime: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace:   metric.Namespace,
+				Name:        "config_last_reload_successfull_timestamp_seconds",
+				Help:        "Timestamp of the last successful configuration reload.",
+				ConstLabels: constLabels,
+			}),
+		reloadOperation: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: metric.Namespace,
+				Name:      "success",
+				Help:      `Cumulative number of Ingress controller reload operations`,
+			},
+			operation,
+		),
+		reloadOperationErrors: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: metric.Namespace,
+				Name:      "errors",
```

Comment:
Can this be changed to something more specific like `nginx_ingress_controller_reload_errors`? Prometheus itself calls this `prometheus_config_last_reload_successful`.

Reply:
Filed this as #2737.